erpnext/projects/doctype/task/task.py

```diff
@@ -89,11 +89,11 @@ def validate_progress(self):
 		if (self.progress or 0) > 100:
 			frappe.throw(_("Progress % for a task cannot be more than 100."))
 
-		if self.progress == 100:
-			self.status = 'Completed'
 
 		if self.status == 'Completed':
 			self.progress = 100
+		else:
+			self.progress = 0
```

Comment:
Why are we setting the progress as 0 if the task is incomplete, intermediate values are not allowed? Also we are removing the auto setting of status if progress is 100%